src/runtime/button-api.js

```diff
@@ -98,20 +100,17 @@ export class ButtonApi {
 
   /**
    * @param {!Element} button
-   * @param {!Object|function()} optionsOrCallback
-   * @param {function()=} opt_callback
+   * @param {!../api/subscriptions.ButtonOptions|function()} optionsOrCallback
+   * @param {function():null=} opt_callback
    * @return {!Element}
    */
   attach(button, optionsOrCallback, opt_callback) {
-    const options =
-        typeof optionsOrCallback != 'function' ?
-        optionsOrCallback : null;
-    const callback = /** @type {function()} */ (
-        (typeof optionsOrCallback == 'function' ? optionsOrCallback : null) ||
-            opt_callback);
+    const options = this.getOptions_(optionsOrCallback);
+    const callback = /** @type {function(Event):boolean} */
```

Comment:
Why does it return boolean? Why is typecast needed here if we have a function to return this function to us?

Reply:
This is due to that fact that the returned `callback` function is passed as a param to `addEventListener` and if not typcasted, throws following error:

```
src/runtime/button-api.js:120: ERROR - Invalid type for parameter 2 of function button.addEventListener.
Expected : EventListener|function(Event):boolean|undefined|null
Found    : function(Event|null):boolean|null

    button.addEventListener('click', callback);
                                     ^^^^^^^^
1 error(s), 0 warning(s)```

Same function is used for `attachSmartButton` but not as an argument to `addEventListener` so needed this typecase.

Reply:
Done.